src/main/java/Duke.java

```diff
@@ -1,10 +1,171 @@
+            }
+            if (!command.contains("/")) {//if there is no '/at'
+                throw new DukeException("Event has no date!");
+            }
+            if (command.indexOf("/") == 6) {
+                throw new DukeException("Event has no description!");
+            }
+            return type;
+        }
+
+
+        if (command.startsWith("todo")) {
+            type = "[T]";
+            if (command.equals("todo")) {
+                throw new DukeException("Todo is empty!");
+            }
+            return type;
+        }
+
+        try {
+            if (command.startsWith("done")) {
+                if (isNumeric(command.substring(5)))
+                    type = "done";
+            }
+        } catch (StringIndexOutOfBoundsException e) {
+            type = "invalidDone";
+            throw new DukeException("Please enter a number after done!");
+        }
+
+        if (command.equals("list")) {
+            return "list";
+        }
+        if (command.equals("bye")) {
+            return "bye";
+        }
+        if (type.equals("invalid")) {
+            throw new DukeException("I do not understand, please enter a valid command!");
+        }
+
+        return type;
+    }
+    /*
+    function to check the command type
+     */
+
+    /*
+    function to check if the command is valid
+     */
+    public static task createTask(String type, int numOfTasks, String command) {
+
+        numOfTasks++;
+        task aTask;
+        if (type.equals("[E]")) {
+            aTask = new event(command, numOfTasks, false, type);
+        } else if (type.equals("[D]")) {
+            aTask = new event(command, numOfTasks, false, type);
+        } else {
+            aTask = new todo(command, numOfTasks, false, type);
+        }
+        System.out.println("Got it, I've added the task:" + "\n" +
+                aTask + "\n" +
+                numOfTasks + " tasks are in the list");
+        return aTask;
+    }
+
+
     public static void main(String[] args) {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("Hello, what can I do for you?");
+
+        int MAX_TASK = 100;
+        task[] tasks = new task[MAX_TASK];
+        /*
+            array to store the tasks
+         */
+        Integer numOfTasks = 0;
+        /*
+            number of tasks currently in the list
+         */
+
+        while (true) {
+            try {
+                String command;
+                Scanner in = new Scanner(System.in);
+                command = in.nextLine();
+                Integer length = command.length();
+                String type = checkType(command);
+            /*
+            type is the type of task, event/to do/deadlines
```

Comment:
Even though a comment is good, trying to maintain this comment might become cumbersome once you start introducing new types into the system. Consider refactoring the type into an enum so that it can be managed easier.
Or better yet, use polymorphism so that you don't need a tag to identify the object